x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/Realms.java

```diff
@@ -224,6 +224,15 @@ public Realm realm(String name) {
         }
         // always add built in first!
         realms.add(0, reservedRealm);
+        Set<String> duplicateNameRealms = new HashSet<>();
+        for (Realm r : realms) {
```

Comment:
We're iterating through realms twice now but this allows us to find and report multiple instances of duplicate names, instead of stopping on first failure. I'm open to suggestions to do this in a more clever way, but it feels clear enough

Reply:
Thanks for the comment Yogesh! The problem with you approach is that 
the error message will be missing one of the duplicate realms in a config every time. i.e. if `my_common_name` is used for 3 realms, the first time we encounter this will iterating, it won't be in the seen list, so we won't add it in the duplicates and then we will only log the other two as having the same realm name. 

I think that a proper error message is better than saving us an iteration of a List of Realms that only happens once, when we build the realm list on startup. WDYT ?

Reply:
I'm OK with iterating twice (well, actually N times because we do a `.stream()...count()`) because the list is going to be short & we only do this on startup.

I would probably choose to go swith @bizybot's suggestion above (which I typed out myself and then realised was basically identical to his) or another option it to just track a `foundDuplicate` flag and then only doing the work of iterating & building the message if that flag is true.
```
        List<Realm> realms = new ArrayList<>();
        Set<String> realmNames = new HashSet<>();
        List<String> kerberosRealmNames = new ArrayList<>();
        boolean hasDuplicateRealm = false;
        for (RealmConfig.RealmIdentifier identifier: realmsSettings.keySet()) {
            if(realmNames.add(identifier.getName()) == false) {
                hasDuplicateRealm = true;
            }
        // ...
        }
        if (hasDuplicateRealm) {
            for (RealmConfig.RealmIdentifier identifier: realmsSettings.keySet()) {
                   // ...
            }
        }  
```

Reply:
Thanks both ! I adopted @bizybot 's suggestion